### PR TITLE
UX: Multichain: Add Account or Hardware Wallet button

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**
Replaces the "Add / Import / Hardware" links with a single button that takes the user to another quick modal to add, import, or connect a hardware wallet.

## **Manual testing steps**

_1.  Open the account menu
_2. Click the "Add account or hardware wallet" button
_3. Click each option, ensure the correct modal displays
_4. Click "back" on each screen, ensure it goes back logically in the nested cycle

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/46655/7967d0ce-870c-4ccb-91bc-3f2e9cf55e79

## **Related issues**

_Fixes https://github.com/MetaMask/MetaMask-planning/issues/1374

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/account-list-menu/account-list-menu.js [hunk 1]
```diff
@@ -202,124 +333,22 @@ export const AccountListMenu = ({ onClose }) => {
...
+              display={Display.Flex}
+            >
+              <ButtonSecondary
+                startIconName={IconName.Add}
+                variant={ButtonVariant.Secondary}
+                style={{ margin: '0 auto' }}
+                onClick={() => setActionMode('menu')}
+                data-testid="multichain-account-menu-popover-action-button"
+              >
+                Add account or hardware wallet
```
Comment (NidhiKJha) [inline ui/components/multichain/account-list-menu/account-list-menu.js:350]: Nit: This should be using translations

File: ui/components/multichain/account-list-menu/account-list-menu.js [hunk 2]
```diff
@@ -202,124 +333,22 @@ export const AccountListMenu = ({ onClose }) => {
...
+            <Box
+              paddingTop={4}
+              paddingBottom={4}
+              alignItems={AlignItems.center}
+              display={Display.Flex}
+            >
+              <ButtonSecondary
+                startIconName={IconName.Add}
+                variant={ButtonVariant.Secondary}
+                style={{ margin: '0 auto' }}
```
Comment (NidhiKJha) [inline ui/components/multichain/account-list-menu/account-list-menu.js:346]: I think we can remove the inline styling here, it should have a paddingLeft and right 16px same as the cell children. Also, the CTA size should be LG
```suggestion
            <Box
              paddingTop={2}
              paddingBottom={4}
              paddingLeft={4}
              paddingRight={4}
              alignItems={AlignItems.center}
              display={Display.Flex}
            >
              <ButtonSecondary
                startIconName={IconName.Add}
                variant={ButtonVariant.Secondary}
                size={ButtonSecondarySize.Lg}
                block
```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Looking really good. Left some comments. Question: I think we should create a separate component for the New modal with add, import and hardware wallet buttons as we are gonna need this modal in connetions flow as well https://www.figma.com/file/LDlI7yjKBNIY60MtQv9QS0/Amon-Hen%2FV1?type=design&node-id=4878-849206&mode=dev. wdyt?

File: ui/components/multichain/account-list-menu/account-list-menu.js (hunk 1 above)
Comment (darkwing) [reply ui/components/multichain/account-list-menu/account-list-menu.js:350]: Good shout!

File: ui/components/multichain/account-list-menu/account-list-menu.js (hunk 2 above)
Comment (darkwing) [reply ui/components/multichain/account-list-menu/account-list-menu.js]: Implemented!

**Review by NidhiKJha — DISMISSED** [review]
LGTM!

File: ui/components/multichain/account-list-menu/account-list-menu.js [hunk 3]
```diff
@@ -138,6 +147,127 @@ export const AccountListMenu = ({ onClose }) => {
...
+                onClick={() => {
+                  trackEvent({
+                    category: MetaMetricsEventCategory.Navigation,
+                    event: MetaMetricsEventName.AccountAddSelected,
+                    properties: {
+                      account_type: MetaMetricsEventAccountType.Imported,
+                      location: 'Main Menu',
+                    },
+                  });
+                  setActionMode('import');
```
Comment (DDDDDanica) [inline ui/components/multichain/account-list-menu/account-list-menu.js:186]: Optional: can we have local const to store the type of actionMode including `add`, `import` and `menu`. so there won't be any typo error when we use them :)

Comment (darkwing) [reply ui/components/multichain/account-list-menu/account-list-menu.js]: Great idea!  Updated!

File: app/_locales/en/messages.json [hunk 4]
```diff
@@ -246,6 +246,9 @@
   "addIPFSGateway": {
     "message": "Add your preferred IPFS gateway"
   },
+  "addImportAccount": {
+    "message": "Add account or hardware wallet"
```
Comment (georgewrmarshall) [inline app/_locales/en/messages.json:250]: This seems like a very long string to have inside of a button just want to make sure that it's been signed off by content cc @coreyjanssen

**Review by NidhiKJha — APPROVED** [review]
LGTM

**Review by vthomas13 — APPROVED** [review]

**Comment by coreyjanssen** [comment]
What exactly is a custodial account? If this is something we're using in lieu of Account Snaps/Snap account, I don't know that it's the best option. Additionally, it doesn't follow our standards for sentence case and should probably have a relevant icon as with the rest of the menu.

File: app/_locales/en/messages.json (hunk 4 above)
Comment (coreyjanssen) [reply app/_locales/en/messages.json:250]: I did sign off on the text but am now doubting it! It seemed like the best option to encompass what lies behind the CTA but perhaps Add account or wallet would suffice? Wdyt?

**Pull request merged**